File: python/maxProfit.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        if len(prices) < 2:
            return 0
        least = prices[0]
        max_profit = 0
        for i0 in prices:
            curr_profit = i0 - least
            if curr_profit > max_profit:
                max_profit = curr_profit 
            if i0 < least:
                least = i0
        return max_profit

File: java/maxProfit.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxProfit = 0;
        
        if (prices.length > 0) {
            int minPrice = prices[0];
            for (int i0 = 1; i0 < prices.length; i0++) {
                minPrice = Math.min(minPrice, prices[i0]);
                maxProfit = Math.max(maxProfit, prices[i0] - minPrice);
            }
        }   
        
        return maxProfit;
    }
}
